lib/SimpleSAML/XML/Signer.php

```diff
@@ -303,8 +303,10 @@ public function sign($node, $insertInto, $insertBefore = null)
             $options
         );
 
-        /** @var \RobRichards\XMLSecLibs\XMLSecurityKey $this->privateKey */
-        $objXMLSecDSig->sign($this->privateKey);
+        /** @var \RobRichards\XMLSecLibs\XMLSecurityKey $privateKey */
```

Comment:
Shouldn't this be fixed in the definition of the property, instead of here?

Reply:
No, it can be `false`, but in this case we've already tested for that here:
https://github.com/simplesamlphp/simplesamlphp/blob/9f97a5dd7c353c4fde3efbf8786577ba69713382/lib/SimpleSAML/XML/Signer.php#L286

I think I can clean this up.. I'll give it a shot

Reply:
Ah, I see! Thanks for clarifying, it's really difficult to review this without looking at the entire code...